cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -400,6 +420,28 @@ public void stop(ProgramId programId, @Nullable String runId) throws Exception {
     return runtimeInfo.getController().stop();
   }
 
+  /**
+   * Save runtime arguments for all future runs of this program. The runtime arguments are saved in the
```

Comment:
Not in this PR, but when we add runtime arguments from UI they do not get persisted. They are only used for the next run. However we do persist the `preferences`. The endpoint `/namespaces/{namespace-id}/apps/{application-id}/{program-type}/{program-id}/preferences` in the `PreferenceHttpHandler` is doing the exact same thing done in this method. 

Should we deprecate `/apps/{app-id}/{program-type}/{program-id}/runtimeargs` endpoint for setting and getting runtimeargs and use the ones from `PreferenceHttpHandler`?